### PR TITLE
[core] Added CCondVar

### PR DESCRIPTION
**Pull request by maxsharabayko**
Condition variable wrapper class

Yet another subset of PR #800.

### CODE REVIEW

File: srtcore/sync.h [hunk 1]
```diff
@@ -267,6 +267,57 @@ class InvertedLock
...
+    ///
+    /// @returns false if the relative timeout specified by rel_time expired,
+    ///          true otherwise (signal or spurious wake up).
+    ///
+    /// @note Calling this function if lock.mutex()
+    /// is not locked by the current thread is undefined behavior.
+    /// Calling this function if lock.mutex() is not the same mutex as the one
+    /// used by all other threads that are currently waiting on the same
+    /// condition variable is undefined behavior.
+    bool wait_for(UniqueLock& lock, const steady_clock::duration& rel_time);
```
Comment (ethouris) [inline srtcore/sync.h]: Question: is wait_until agreed to be implemented only in the `CSync` class? Still acceptable, but maybe, you know.

File: srtcore/sync.cpp [hunk 2]
```diff
@@ -382,6 +382,53 @@ srt::sync::Mutex* srt::sync::UniqueLock::mutex()
...
+}
+
+srt::sync::CCondVar::~CCondVar()
+{
+    pthread_cond_destroy(&m_cv);
+}
+
+bool srt::sync::CCondVar::wait_for(UniqueLock& lock, const steady_clock::duration& rel_time)
+{
+    return (SyncEvent::wait_for(&m_cv, &lock.mutex()->ref(), rel_time) != ETIMEDOUT);
```
Comment (ethouris) [inline srtcore/sync.cpp]: I think it makes much more sense to make `SyncEvent::wait_for` expect `srt::sync::Mutex` argument - just to reconsider.

Comment (maxsharabayko) [reply srtcore/sync.cpp]: UniqueLock is to align with the [C++11 cv](https://en.cppreference.com/w/cpp/thread/condition_variable/wait_for).
Although the class cannot be mapped directly co std::condition_variable at the moment due to the return of this `wait_for()` function.

Comment (ethouris) [reply srtcore/sync.cpp]: We'll make do. This can be later changed into the aliased or crafted type of `cv_result`.

File: srtcore/sync.h (hunk 1 above)
Comment (maxsharabayko) [reply srtcore/sync.h]: Added.

**Comment by maxsharabayko** [comment]
Post-review changes ^^^

**Comment by maxsharabayko** [comment]
TODO:
- [ ] Add monotonic version.

**Comment by ethouris** [comment]
Please don't use `CLOCK_MONOTONIC` as a parameter name. Imageine what happens when the preprocessor resolves it in case when it's provided from the header as
```
#define CLOCK_MONOTONIC 1
```
I changed in my version already to `IS_CLOCK_MONOTONIC`

**Comment by maxsharabayko** [comment]
Build fixes ^^^

**Pull request merged**